enhancements/olm-admission-webhooks.md

```diff
@@ -114,6 +114,29 @@ The certificate creation and lifecycle management is planned to utilize the same
 
 There is a great deal of community support for integration with certificate management projects such as the [service-ca operator](https://github.com/openshift/service-ca-operator) or [cert-manager](https://github.com/jetstack/cert-manager). Cluster Admins should be given the opportunity to integrate with either of these tools be disabling the OLM management of certs and adding appropriate annotations to the deployment and webhook resources via the [SubscriptionConfig resource](https://github.com/operator-framework/operator-lifecycle-manager/blob/8985872bfd5888c8d7d49e4dfc9be162a87691e7/pkg/api/apis/operators/v1alpha1/subscription_types.go#L41-L87).
 
+### Deployment Installation Order
+
+There are a number of customer usecases in which the order that deployments are rollout matters. Consider the following:
+
+- An operator may be written to assume that any CR it sees has passed through a webhook defined in its CSV. In this case, OLM should provide the Operator Author with the ability to install the admission webhook before the operator to prevent the operator from reconciling CRs that haven't gone through the webhook.
+- An operator may be written such that it expects to run some custom logic prior to starting the webhook. This logic may involve creating a CR that has special values that would be changed by the webhook defined in the CSV.
+
+Both of the usecases described above can be addressed by updating OLM to:
+
+- Create Deployments defined in the CSV in order.
+- When creating the deployments defined in the CSV:
+  - Wait for the Deployment to reach the `READY` state.
+  - OLM should check if the deployment owns any webhook or APIServices resources and if so, create these resource.
+  - Create the next deployment in the CSV.
+
+### Webhook Upgrades
+
+OLM must support upgrading webhooks between versions of a CSV. OLM should be updated to handle the following usecases:
+
+- If an admission webhook is removed in a new version of the CSV, the on cluster Webhook should be deleted.
+- If a new webhook is added in a new version of the CSV, the webhook should be deployed on cluster.
+- If the webhook is updated in a new version of the CSV, the deployment that implements the old version of the CSV should be removed, and existing resources should be migrated to the new deployment that implements the webhook.
```

Comment:
There is some nuance here - 

Deployments that have the same name are `update`d so that pods from the old replicaset stick around long enough to handoff to pods in the new replicaset (i.e. for leader election).

If the deployment name has changed, we just delete/create.

But my bigger question is: are there cases where we would need to temporarily disable a webhook for upgrade?